runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -12121,7 +12207,52 @@ J9::Power::CodeGenerator::inlineDirectCall(TR::Node *node, TR::Register *&result
 
          if ((node->isUnsafeGetPutCASCallOnNonArray() || !TR::Compiler->om.canGenerateArraylets()) && node->isSafeForCGToFastPathUnsafeCall())
             {
-            resultReg = VMinlineCompareAndSwapObject(node, cg);
+            resultReg = VMinlineCompareAndSetOrExchangeReference(node, cg, false);
+            return true;
+            }
+         break;
+
+      case TR::jdk_internal_misc_Unsafe_compareAndExchangeInt:
+        if ((node->isUnsafeGetPutCASCallOnNonArray() || !TR::Compiler->om.canGenerateArraylets()) && node->isSafeForCGToFastPathUnsafeCall())
+            {
+            if (disableCAEIntrinsic)
+               {
+               break;
+               }
+            resultReg = VMinlineCompareAndSetOrExchange(node, cg, 4, true);
+            return true;
```

Comment:
May I suggest eliminating the `break` in the middle of this case block by changing this to the following?

```
if (!disableCAEIntrinsic)
   {
   resultReg = VMinlineCompareAndSetOrExchange(node, cg, 4, true);
   return true;
   }
```

and similarly for the other new cases in this `switch`?

Reply:
I'll do this.